src/txorphanage.h

```diff
@@ -50,7 +50,12 @@ class TxOrphanage {
     /** Erase all orphans included in or invalidated by a new block. Returns wtxids of erased txns. */
     std::vector<uint256> EraseForBlock(const CBlock& block) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
 
-    /** Limit the orphanage to the given maximum */
+    /** Limit the orphanage to the given maximum. Delete orphans whose expiry has been reached.
```

Comment:
I think the expiry orphans could be announced at the protocol-level, as part of the “package-link”, i.e somewhere in `sendpackages` or in `ancpkinfo`. Those limits could be announced from the peer’s `PeerManagerImpl` with a push-interface (e.g the `ZMQNotificationInterface` ), that way a Lightning node could adapt its package broadcasting strategy accordingly.